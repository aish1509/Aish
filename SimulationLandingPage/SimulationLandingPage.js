import React from 'react';
import {DataStore} from '@shared/core';
import {Layout} from '@math/m180';
import Header from '../../utils/Header';
import history from '../../utils/history';
import {SimulationC1B1} from '../../C1B1';

const SimulationLandingPage = props => {
	const loading = () => <div>loading...</div>;
	const pathname = props.match.path;

	const ready = profile => (
		<div
			style={{
				display: 'flex',
				flexDirection: 'column',
				height: '85vh',
				position: 'relative'
			}}
		>
			<SimulationC1B1 path={pathname} history={history} />
		</div>
	);
	const page = ({profile}) => (profile ? ready(profile) : loading());

	return (
		<Layout>
			<Header pathname={pathname} />
			<DataStore.Subscribe>
				{({state}) => page(state)}
			</DataStore.Subscribe>
		</Layout>
	);
};

export default SimulationLandingPage;
