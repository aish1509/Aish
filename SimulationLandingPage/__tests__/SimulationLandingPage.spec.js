import React from 'react';
import {render, cleanup, waitForElement} from 'react-testing-library';
import 'jest-dom/extend-expect';

import {M180Theme} from '@math/common';
import {DataStore} from '@shared/core';

import SimulationLandingPage from '../index';

describe('<SimulationLandingPage />', () => {
	let mockMatch = null;

	let mockProfile = {
		username: 'username',
		scopes: 'scopes',
		app: 'm180'
	};

	afterEach(cleanup);
	it('SimulationLandingPage page is loading', () => {
		DataStore.set('profile', null);
		mockMatch = {path: ''};
		const {getByText} = render(
			<DataStore.Provider>
				<M180Theme>
					<SimulationLandingPage match={mockMatch} />
				</M180Theme>
			</DataStore.Provider>
		);
		expect(getByText('loading...')).toBeDefined();
	});
	it('SimulationLandingPage renders content correctly', () => {
		DataStore.set('profile', mockProfile);
		mockMatch = {path: ''};
		const {container} = render(
			<DataStore.Provider>
				<M180Theme>
					<SimulationLandingPage match={mockMatch} />
				</M180Theme>
			</DataStore.Provider>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
