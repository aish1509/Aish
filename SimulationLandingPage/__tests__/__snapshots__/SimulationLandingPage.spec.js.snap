// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimulationLandingPage /> SimulationLandingPage renders content correctly 1`] = `
<div
  style="display: flex; flex-direction: column;"
>
  <div
    class="makeStyles-container-9"
  >
    <div
      class="makeStyles-leftSquare-12"
    >
      <div
        class="makeStyles-svgPos-11"
      >
        <svg>
          m180logo.svg
        </svg>
      </div>
      <div
        class="makeStyles-triangleTopleft-13"
      />
    </div>
    <div
      class="makeStyles-headerText-10"
    >
      MATH 180
    </div>
    <div
      class="makeStyles-container-14"
    >
      Welcome, 
      firstname
      !
      <button
        class="makeStyles-logout-16"
        type="button"
      >
        Logout
      </button>
    </div>
  </div>
  <div>
    loading...
  </div>
</div>
`;
